src/pipeline/tests/simple_extract.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod common;
+
+use api::v1::value::ValueData;
+use api::v1::{ColumnDataType, ColumnSchema, SemanticType};
+use lazy_static::lazy_static;
+
+lazy_static! {
+    static ref EXPECTED_SCHEMA: Vec<ColumnSchema> = vec![
+        common::make_column_schema(
+            "commit_author".to_string(),
+            ColumnDataType::String,
+            SemanticType::Field,
+        ),
+        common::make_column_schema(
+            "greptime_timestamp".to_string(),
+            ColumnDataType::TimestampNanosecond,
+            SemanticType::Timestamp,
+        ),
+    ];
+}
+
+#[test]
+fn test_gsub() {
+    let input_value_str = r#"
+    [
+      {
+        "commit": {
+            "commitTime": "1573840000.000",
+            "commitAuthor": "test"
+        }
+      }
+    ]
+"#;
+
+    let pipeline_yaml = r#"
+---
+processors:
+  - simple_extract:
+      field: commit, commit_author
+      key: "commitAuthor"
```

Comment:
Can we check again this UI? if I understand correctly it means we use the first field as input, apply the processor "commitAuthor" and use the second field as output. 

If this follows how we have been using "field" it's ok.

Reply:
Yes, it's tricky at first sight. In the `field`, the first key is the name of the input object, currently there is no way to point to the 'root context'. The second key represents the output's name; if not provided, it uses the input's name, which leads to a replacement of the original object. cc @paomian